lib/mix/tasks/schema.ex

```diff
@@ -34,7 +43,16 @@ defmodule Mix.Tasks.Ecto.Ch.Schema do
 
     statement = "select name, type from system.columns " <> where
 
-    conn = connect(_config = [])
+    repos = Mix.Ecto.parse_repo(kvs)
+
+    config =
+      Enum.find_value(repos, fn repo ->
+        Mix.Ecto.ensure_repo(repo, kvs)
+        repo.__adapter__() == Ecto.Adapters.ClickHouse
```

Comment:
This way it worked fine for me. Else it would take my first configured repo.
```suggestion
if repo.__adapter__() == Ecto.Adapters.ClickHouse do
  repo.config()
end
```

Reply:
Can you please explain this change?